colossalai/zero/sharded_param/tensorful_state.py

```diff
@@ -8,22 +8,22 @@ class TensorState(Enum):
     HOLD = 1
     HOLD_AFTER_FWD = 2
     HOLD_AFTER_BWD = 3
+    COMPUTE = 4
 
 
 class StatefulTensor(object):
-    """A Structure stores a Torch Tensor and labeled states.
-
+    """A Structure stores a Torch Tensor and labeled states. 
+    Inspired from the paper:
     PatrickStar: Parallel Training of Pre-trained Models via Chunk-based Memory Management
 
     https://arxiv.org/abs/2108.05818
     """
 
-    def __init__(self, tensor: torch.Tensor, state: TensorState = TensorState.HOLD) -> None:
+    def __init__(self, tensor: torch.Tensor, state: Optional[TensorState] = TensorState.HOLD) -> None:
```

Comment:
When is the `state` `None`?

Reply:
But `state: Optional[TensorState]` means `state` can be `TensorState` or `None`. When can it be `None`?

Reply:
OK, I misuse the Optional. remove it